test/integration/src/test/java/io/pravega/test/integration/StreamSegmentContainerMetricsTest.java

```diff
@@ -0,0 +1,195 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static io.pravega.shared.MetricsTags.containerTag;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+@Slf4j
+public class StreamSegmentContainerMetricsTest {
+
+    private final ScalingPolicy scalingPolicy = ScalingPolicy.fixed(1);
+    private final StreamConfiguration config = StreamConfiguration.builder().scalingPolicy(scalingPolicy).build();
+    private TestingServer zkTestServer = null;
+    private PravegaConnectionListener server = null;
+    private ControllerWrapper controllerWrapper = null;
+    private Controller controller = null;
+    private StatsProvider statsProvider = null;
+    private ServiceBuilder serviceBuilder = null;
+    private AutoScaleMonitor monitor = null;
+
+    @Before
+    public void setup() throws Exception {
+        final int controllerPort = TestUtils.getAvailableListenPort();
+        final String serviceHost = "localhost";
+        final int servicePort = TestUtils.getAvailableListenPort();
+        final int containerCount = 4;
+
+        // 1. Start Metrics service
+        log.info("Initializing metrics provider ...");
+
+        MetricsConfig metricsConfig = MetricsConfig.builder()
+                .with(MetricsConfig.ENABLE_STATSD_REPORTER, false)
+                .build();
+        metricsConfig.setDynamicCacheEvictionDuration(Duration.ofSeconds(60));
+
+        MetricsProvider.initialize(metricsConfig);
+        statsProvider = MetricsProvider.getMetricsProvider();
+        statsProvider.startWithoutExporting();
+        log.info("Metrics Stats provider is started");
+
+        // 2. Start ZK
+        this.zkTestServer = new TestingServerStarter().start();
+
+        // 3. Start Pravega SegmentStore service.
+        serviceBuilder = ServiceBuilder.newInMemoryBuilder(ServiceBuilderConfig.getDefaultConfig());
+        serviceBuilder.initialize();
+        StreamSegmentStore store = serviceBuilder.createStreamSegmentService();
+        monitor = new AutoScaleMonitor(store, AutoScalerConfig.builder().build());
+        TableStore tableStore = serviceBuilder.createTableStoreService();
+
+        this.server = new PravegaConnectionListener(false, false, "localhost", servicePort, store, tableStore,
+                monitor.getStatsRecorder(), monitor.getTableSegmentStatsRecorder(), new PassingTokenVerifier(),
+                null, null, true);
+        this.server.startListening();
+
+        // 4. Start Pravega Controller service
+        this.controllerWrapper = new ControllerWrapper(zkTestServer.getConnectString(), false,
+                controllerPort, serviceHost, servicePort, containerCount);
+        this.controllerWrapper.awaitRunning();
+        this.controller = controllerWrapper.getController();
+    }
+
+    @After
+    public void tearDown() throws Exception {
```

Comment:
I suggest you invoke `shutdown` here in case something in your test fails before it reaches the `shutdown` call.

Reply:
After thinking it over, this straightforward change doesn't deserve a heavy integration test, so I replaced it with a lightweight unit test. Please review, thanks